meinberlin/apps/budgeting/assets/__tests__/EndSessionLink.jest.jsx

```diff
@@ -0,0 +1,38 @@
+import React from 'react'
+import { render, screen, fireEvent, waitFor } from '@testing-library/react'
+import { EndSessionLink } from '../EndSessionLink'
+
+test('End session - check: modal rendered', async () => {
+  render(<EndSessionLink />)
+
+  const modalBtn = screen.getByRole('button', { name: 'End session' })
+  const modalRender = screen.queryByRole('dialog')
+  expect(modalRender).toBeNull()
+  fireEvent.click(modalBtn)
+
+  await waitFor(() => {
+    const modalRender = screen.findByRole('dialog')
+    expect(modalRender).toBeTruthy()
+  })
+})
+
+test('End session - check: end session url fetch failed', async () => {
+  // overwrite global.fetch with mock function
+  global.fetch = jest.fn().mockRejectedValue('testing: expected network error')
+
+  render(<EndSessionLink />)
+  // toggle modal
+  const modalBtn = screen.getByRole('button', { name: 'End session' })
+  fireEvent.click(modalBtn)
+
+  // get click 'End Session' btn in modal (2nd instance of btn)
+  const endSessionBtn = await screen.getAllByText('End session', { selector: 'button' })[1]
```

Comment:
@goapunk do you know why`find/query/getAllByRole('button', { name: 'End session' })[1]` wouldn't work and returns undefined when console.logged? it's more a point of annoyance but role is a good option to check if something is there

Reply:
It works if you also include the  hidden option:
```
screen.getAllByRole('button', {name:'End session', hidden: true})[1]   
```

Reply:
the `await` here is unnecessary, it's only needed if you use `findBy*`

Reply:
@goapunk so if it is only finding the hidden button and not rendering the modal and finding the button we should delete this right? my understanding is that a test should mimic user behavior, and a user cannot click a hidden button therefore we shouldn't have the test?